pom.xml

```
@@ -340,11 +350,19 @@
                   </module>
                   <module name="RegexpSinglelineJava">
                     <property name="format" value="junit[.]framework[.]TestCase" />
-                    <property name="message" value="Use JUnit4+ @Test annotation instead of TestCase" />
+                    <property name="message" value="Use JUnit5+ @Test annotation instead of TestCase" />
+                  </module>
+                  <module name="RegexpSinglelineJava">
+                    <property name="format" value="org[.]junit[.][^jupiter]" />
+                    <property name="message" value="Use JUnit5 (JUnit Jupiter) instead of JUnit4 (or lower)" />
```

I don't think that's the correct regex. That is a character set negation, not a negative lookaround. I think you're trying to go for:

```suggestion
                    <property name="format" value="org[.]junit[.](?!jupiter)" />
                    <property name="message" value="Use JUnit5 (JUnit Jupiter) instead of JUnit4 (or lower)" />
```

However, I don't know if that works.

---

Addressed in 10a1efd